cmds/boot/boot.go

```diff
@@ -0,0 +1,403 @@
+//	The first bootable device found in the block device tree is the one used
+//	Windows is not supported (that is a work in progress)
+//
+// Example:
+//	boot -v 	- Start the script in verbose mode for debugging purpose
+
+package main
+
+import (
+	"bufio"
+	"encoding/binary"
+	"fmt"
+	"github.com/u-root/u-root/pkg/kexec"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"flag"
+	"syscall"
+)
+
+const (
+	bootableMBR = 0xaa55
+	signatureOffset = 510
+)
+
+var verbose bool
+type options struct {
+        verbose bool
+}
+
+func check(e error) {
+	if e != nil {
+		log.Fatal(e)
+	}
+}
+
+func blkDevicesList(blkpath string, devpath string) []string {
+	var blkDevices []string
+	files, err := ioutil.ReadDir(blkpath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	for _, file := range files {
+		deviceEntry, err := ioutil.ReadDir(blkpath + file.Name() + devpath)
+		if err != nil {
+			if ( verbose ) {
+				println("can t read directory")
+			}
+			continue
+		}
+		blkDevices = append(blkDevices, deviceEntry[0].Name())
+	}
+	return (blkDevices)
+}
+
+// checkForBootableMBR is looking for bootable MBR signature 
+// Current support is limited to Hard disk devices and USB devices
+func checkForBootableMBR(path string) (bool,error) {
+	var sig uint16
+	f, err := os.Open(path)
+	if err != nil {
+		return false, err
+	}
+	if err := binary.Read(io.NewSectionReader(f, signatureOffset, 2), binary.LittleEndian, &sig); err != nil {
+		return false,err
+	}
+	if sig != bootableMBR {
+		return false, nil
+	}
+	return true,nil	
+}
+
+// getDevicePartList returns all devices attached to a specific name like /dev/sdaX where X can move from 0 to 127
+// FIXME no support for devices which are included into subdirectory within /dev
+func getDevicePartList(path string) []string {
+	var returnValue []string
+	files, err := ioutil.ReadDir("/dev/")
+	if err != nil {
+		log.Fatal(err)
+	}
+	for _, file := range files {
+		if strings.HasPrefix(file.Name(), path) {
+			// We shall not return full device name
+			if file.Name() != path {
+				// We shall check that the remaining part is a number
+				returnValue = append(returnValue, file.Name())
+			}
+		}
+	}
+	return returnValue
+}
+
+// getSupportedFilesystem returns all block file system supported by the linuxboot kernel
+func getSupportedFilesystem() []string {
+	var returnValue []string
+	var err error
+	file, err := os.Open("/proc/filesystems")
+	check(err)
+	scanner := bufio.NewScanner(file)
+	if err = scanner.Err(); err != nil {
+		log.Fatal(err)
+	}
+
+	for scanner.Scan() {
+		fields := strings.Fields(scanner.Text())
+		if fields[0] == "nodev" {
+			continue
+		}
+		if fields[0] != "" {
+			returnValue = append(returnValue, fields[0])
+    		}
+	}
+
+	err=file.Close()
+	check(err)
+	return returnValue
+
+}
+
+// mountEntry tries to mount a specific block device
+func mountEntry(path string, supportedFilesystem []string) bool {
+	var returnValue bool
+	var err error
+	err=syscall.Mkdir("/u-root", 0777)
```

Comment:
use ioutil.TempDir, I'm getting errors as this file exists. 
Or use os.IsExist(err) and ignore the error if so.

Reply:
Should be fixed in latest commit